### ISSUE TITLE
I would be nice to have ethernet only infiniband

### ISSUE THREAD

**Issue by dswartz** [astersmith/esos#113]
Some HCA (such as connectx-2) can function in either IB or enet mode.  In some cases, the mlx4_en driver works either way, and can be toggled back and forth.  In others, it seems like you build an enet only version of mlx4_en (the 1.9 series of mlx_en for vsphere 5.x works this way.)  This would be very handy, since you could then plug a random mellanox HCA in and get a (much) cheaper 10gbe enet.

**Comment by msmith626** [comment]
This isn't something you can configure with '/etc/infiniband/openib.conf' already? If not, can you provide the specifics of what module is needed, or what isn't? Or if there is a different kernel configuration option needed.

--Marc

**Comment by dswartz** [comment]
I need to do some research.  Apparently there are two ways of doing this:

1. Have an mlx4_en driver that goes both ways.  You then have a program that you run to toggle it.
2. Have an mlx4_en driver that is only built for one mode.  This is what vsphere did when they went to the recent 1.9.? drivers.  e.g. all they support is 10gbe.  How do you build the mlx4* modules?

**Comment by dswartz** [comment]
http://www.mellanox.com/pdf/products/software/MLNX_EN_Linux_README.txt

**Comment by dswartz** [comment]
Any thoughts?

**Comment by msmith626** [comment]
Sorry for the delay Dan. I see this for the kernel configuration:
$ cat misc/linux-3.14.36.config | grep -i mlx4
CONFIG_MLX4_EN=y
CONFIG_MLX4_EN_DCB=y
CONFIG_MLX4_CORE=y
CONFIG_MLX4_DEBUG=y
CONFIG_MLX4_INFINIBAND=m

I honestly haven't read the above article, but it sounds like there are 2
drivers for the card... if you just don't load the IB driver for the card
(eg, disable rc.openib in /etc/rc.conf) does it work? I'll read more about
this tonight or tomorrow, promise. For the tool that you can use to toggle
it, where do you get that? Is it OSS?


--Marc

**Comment by dswartz** [comment]
It seems to depend which OFED package you pick up.  I know it didn't 'just work' with the connectx-2 cards.  On the other hand, I scored a new connectx-3 card on ebay for fairly cheap.  I plugged it in, and it printed some messages about converged ethernet cards, and guess what?  10gbe ethernet for me!  Maybe the newer cards present two different identities to the mlx4_en driver without needing any hacking?  So, since you have more than enough on your plate, I will close this :)
